ui/components/src/Table/VirtualizedTable.tsx

```diff
@@ -0,0 +1,179 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Column, HeaderGroup, Row, flexRender } from '@tanstack/react-table';
+import { Box } from '@mui/material';
+import { TableVirtuoso, TableComponents, TableVirtuosoHandle, TableVirtuosoProps } from 'react-virtuoso';
+import { useRef, useMemo } from 'react';
+import { TableRow } from './TableRow';
+import { TableBody } from './TableBody';
+import { InnerTable } from './InnerTable';
+import { TableHead } from './TableHead';
+import { TableCell, TableCellProps } from './TableCell';
+import { VirtualizedTableContainer } from './VirtualizedTableContainer';
+import { TableProps } from './model/table-model';
+import { useVirtualizedTableKeyboardNav } from './hooks/useVirtualizedTableKeyboardNav';
+
+type TableCellPosition = {
+  row: number;
+  column: number;
+};
+
+export type VirtualizedTableProps<TableData extends Record<string, unknown>> = Required<
+  Pick<TableProps<TableData>, 'height' | 'width' | 'density'>
+> & {
+  onRowClick: (id: string) => void;
+  rows: Array<Row<TableData>>;
+  columns: Array<Column<TableData, unknown>>;
+  headers: Array<HeaderGroup<TableData>>;
+};
```

Comment:
Doing it like this because I don't think you can extend generics and get all the types working properly.